### ISSUE TITLE
Should use newer version of DPDK 

### ISSUE THREAD

**Issue by thearossman** [stanford-esrg/retina#48]
DPDK LTS releases 21.11+ contain breaking API changes that prevent compilation. At this point, 21.08 is ~3 years old, and DPDK typically stops maintaining LTS's after 3 years.

**Comment by HKalbasi** [comment]
Would you accept a PR to update DPDK to 23.11 which is the latest LTS? Or do you want to keep supporting the 21.08 and support multiple DPDK versions at the same time?

**Comment by thearossman** [comment]
An upgrade would be AMAZING. This is one of those things we've bookmarked but not prioritized. 

I guess it could be nice to have 21.08 support via a feature flag (or similar) so that folks who are already using Retina could keep developing without redoing a DPDK setup? But definitely only worth supporting both if it's relatively easy to do.

**Comment by HKalbasi** [comment]
I have a working draft on 23.11, will clean it up and make a PR after #75

Supporting 21.08 at the same time is possible, but I guess over time it will become a maintainance burden, specially if we want to add more versions later.

**Comment by tbarbette** [comment]
Typically DPDK project support the latest few versions with #ifdefs. For research it avoids having different DPDK versions for every projects and allow to compare systems more easily. If it's not too much work I'd advise to try to keep some backward compatibility.
